src/chunk.c

```diff
@@ -2640,10 +2640,12 @@ ts_chunk_get_by_id(int32 id, bool fail_if_not_found)
 
 /*
  * Number of chunks created after given chunk.
- * If chunk2.id > chunk1.id then chunk2 is created after chunk1
+ * If chunk2.id > chunk1.id then chunk2 is created after chunk1.
```

Comment:
I did not check all the surrounding code here but this logic seems flawed. Do we really care about chunk ids here or do we actually care about relative position on the time dimension. I think we are looking for the latter and in that case chunk id being greater does not mean that a chunk time is after a chunk with a lower id as chunks might be created by backfilling.

Reply:
I'm not sure what we have to consider 'historical/current' to estimate the fill factor -- current in 'timeseries time' or current in 'real time' (chunk id). In the usual case where we are inserting the current time series data, these two are the same. If we are backfilling, maybe 'current in real time' makes more sense because these are the chunks we created last, and they are likely to have smaller fill factor. @erimatnor maybe you remember the logic behind this choice? Anyway, I think this PR doesn't have to change this logic. Whatever chunk order we use, we don't have to go through all of them.

Reply:
AFAIK, the logic is not flawed. The way this is used is to find a "stable" chunk, i.e., its size is more or less stable and won't see a lot of writes. So, we care about how long ago the chunk was created in logical time, assuming that the more chunks have been created after this one, the more stable it is.

Usually, logical time correlates well with the real time dimension as well, but we also have to account for things like multi-dimensional partitioning, so we could have many chunks with the same or similar time ranges. I think the number of space partitions should be used to set the limit of how far we look beyond the current chunk.

So, if we have 4 space partitions, then one can assume that we should at least look 5 chunks back in logical time (including the current chunk) to find a reasonably stable chunk. I don't think this has to be exact, it is just used for estimation.

Reply:
Note that the number of space partitions could have changed, so there might have been a different number of space partitions when the chunk was created. Not sure how to easily account for this though.

Reply:
I realized another problem with chunk ids -- they are global, for all hypertables. This function looks them up w/o any filter by hypertable id, so all these comparisons with numbers of dimensions etc don't really make sense unless there is only one hypertable in the database. We don't have an index on (hypertable_id, chunk_id), so there is no fast way to look up the chunks for a particular hypertable.

Considering that this logic is a fallback for missing ANALYZE data, all this looks unnecessarily complicated. Couldn't we just treat all the chunks w/o statistics as recent ones, and estimate a fill factor of 0.5 for them, or something?